src/main/java/fr/inria/corese/w3c/junit/dynamic/executor/impl/RdfCanonicalEvaluationTestExecutor.java

```diff
@@ -0,0 +1,194 @@
+ * This executor performs the RDFC-1.0 canonicalization algorithm on an input RDF model
+ * and compares the resulting canonical N-Quads set with the expected canonical N-Quads set.
+ */
+public class RdfCanonicalEvaluationTestExecutor implements TestExecutor {
+
+    private static final Logger logger = LoggerFactory.getLogger(RdfCanonicalEvaluationTestExecutor.class);
+
+    // Base URL for W3C RDF Canonicalization tests
+    private static final String W3C_BASE_URL = "https://w3c.github.io/rdf-canon/tests/";
+    private static final String TEST_SUBDIR = "rdfc10";
+    /**
+     * Constructs a new RdfCanonicalEvaluationTestExecutor.
+     */
+    public RdfCanonicalEvaluationTestExecutor() {
+        // No initialization required
+    }
+
+    /**
+     * Executes the RDF Canonicalization evaluation test case.
+     * The process involves loading the input (action) model, canonicalizing it,
+     * loading the expected (result) model, and comparing the resulting N-Quads statements.
+     *
+     * @param testCase The W3cTestCase containing the test metadata (URIs, name).
+     * @throws Exception If any error occurs during file loading, parsing, or canonicalization.
+     * @throws AssertionError If a StackOverflowError is detected during canonicalization, indicating cyclic structures.
+     */
+    @Override
+    public void execute(W3cTestCase testCase) throws Exception {
+        String testName = testCase.getName();
+        URI actionFileUri = testCase.getActionFileUri();
+        URI resultFileUri = testCase.getResultFileUri();
+
+        logger.info("Executing RDF Canonical evaluation test: {}", testName);
+
+        try {
+            // STEP 1: Resolve and load the action and result files
+            String actionFilePath = resolveAndLoadFile(actionFileUri);
+            String resultFilePath = resolveAndLoadFile(resultFileUri);
+
+            // STEP 2: Create and populate the action model from the input file
+            Model actionModel = loadModelFromFile(actionFilePath);
+
+
+            // STEP 3: Canonicalize the action model using RDFC-1.0
+            Model canonicalizedModel = canonicalize(actionModel);
+
+            // STEP 4: Load the expected canonical result into a model for comparison
+            Model expectedModel = loadModelFromFile(resultFilePath);
+
+
+            // STEP 5: Compare models by CONTENT (Set of N-Quads strings)
+            // Convert canonicalized statements to a Set of N-Quads strings
+            Set<String> canonicalizedNQuads = new HashSet<>();
+            for (Statement stmt : canonicalizedModel.stream().toList()) {
+                canonicalizedNQuads.add(StatementUtils.toNQuad(stmt));
+            }
+
+            // Convert expected statements to a Set of N-Quads strings
+            Set<String> expectedNQuads = new HashSet<>();
+            for (Statement stmt : expectedModel.stream().toList()) {
+                expectedNQuads.add(StatementUtils.toNQuad(stmt));
+            }
+
```

Comment:
Isn't the comparison missing in this test?
canonicalizedNQuads and expectedNQuads are both built, but they’re never compared, so the test will always pass unless an exception is thrown. A final equality check should be added.


```suggestion
            if (!canonicalizedNQuads.equals(expectedNQuads)) {
                throw new AssertionError("Canonicalization mismatch for test " + testName + ".\nExpected: " + expectedNQuads + "\nActual: " + canonicalizedNQuads);
            }

```